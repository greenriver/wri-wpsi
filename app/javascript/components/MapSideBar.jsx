import React from 'react'
import { withRouter } from 'react-router-dom'
import { Icon, LegendItemButtonRemove } from 'vizzuality-components'
import LayerCard from 'components/LayerCard'
import Switch from 'react-switch'
import defaultButtonStyle from './styles/default_button'

const MapSideBar = ({
  history,
  maskLayer,
  activeLayers,
  selectedRegion,
  onRemoveLayer,
  onToggleLayer,
}) => {

  const sideBarStyles = {
    display: 'flex',
    flexDirection: 'column',
    flex: '1 1 auto',
  }

  const headerStyle = {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: '15px',
    borderBottom: '1px solid #B6C6BC',
  }

  const buttonStyle = {
    ...defaultButtonStyle(),
  }

  const iconStyle = {
    fill: '#FFFFFF',
    backgroundColor: '#003F6A',
    height: '20px',
    width: '20px',
    marginRight: '8px',
  }

  const sideBarContent = {
    padding: '15px 15px 15px 0px',
    flex: '1 1 auto',
    overflow: 'auto',
  }

  const renderRegionInfo = (region) => {
    return (
      <div style={headerStyle}>
        <i>
          {region.name_2 && `${region.name_2} ${region.engtype_2}, `}
          {region.name_1 && `${region.name_1}, `}
          {region.name_0}
        </i>
      </div>
    )
  }

  const renderMaskLayerCard = (layer) => (
    <LayerCard
      variant='simple'
      layer={layer}
      secondaryAction={
        <Switch
          onChange={() => onToggleLayer({ layer })}
          checked={activeLayers.includes(layer)}
          onColor={'#003F6A'}
          offColor={'#B6C6BC'}
          checkedIcon={false}
          uncheckedIcon={false}
          className='square-switch'
        />
      }
    />
  )

  const renderLayerCard = (layer) => (
    <LayerCard
      key={layer.id}
      layer={layer}
      onRemoveLayer={onRemoveLayer}
      secondaryAction={
        <LegendItemButtonRemove
          onRemoveLayer={() => onRemoveLayer(layer)}
          tooltipText='Hide'
        />
      }
    />
  )

  return (
    <div id='sidebar' style={sideBarStyles}>
      <div style={headerStyle}>
        <h1 style={{marginBottom: 0}}>Investigation</h1>

        <button style={buttonStyle} onClick={() => history.push('/map/datasets/water')}>
          <Icon name="icon-plus" style={iconStyle} />
          Add datasets to investigation
        </button>
      </div>

      {selectedRegion && renderRegionInfo(selectedRegion)}

      {maskLayer && renderMaskLayerCard(maskLayer)}

      <div style={sideBarContent}>
        {
          activeLayers
            .filter(layer => layer.id != maskLayer.id)
            .map(layer => renderLayerCard(layer))
        }
      </div>
    </div>
  )
}

import PropTypes from 'prop-types'
MapSideBar.propTypes = {
  history: PropTypes.object.isRequired,
  setModalOpen: PropTypes.func.isRequired,
  onRemoveLayer: PropTypes.func.isRequired,
  onToggleLayer: PropTypes.func.isRequired,
  maskLayer: PropTypes.object,
  activeLayers: PropTypes.array.isRequired,
  selectedRegion: PropTypes.object,
}

export default withRouter(MapSideBar)
