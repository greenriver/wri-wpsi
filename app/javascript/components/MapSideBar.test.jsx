import React from 'react'
import MapSideBar from 'components/MapSideBar'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const component = renderer.create(
    <MapSideBar
      activeLayers={[]}
      setModalOpen={() => null}
      onRemoveLayer={() => null}
    />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
