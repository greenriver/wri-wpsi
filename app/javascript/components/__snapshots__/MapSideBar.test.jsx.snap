// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MapSideBar-sideBar-0-1-1"
  id="sideBar"
>
  <header
    className="MapSideBar-header-0-1-2"
  >
    <h1
      className="MapSideBar-headerTitle-0-1-3"
    >
      Investigation
    </h1>
    <button
      className="MapSideBar-addLayerButton-0-1-7"
      onClick={[Function]}
      style={Object {}}
    >
      <i
        className="icon__plus-circle MapSideBar-addLayerButtonIcon-0-1-8"
      />
      Add datasets
    </button>
  </header>
  <div
    className="LayerCard-container-0-1-10 LayerCard-toggle-0-1-11"
  >
    <header
      className="LayerCard-header-0-1-13"
    >
      <h2
        className="LayerCard-title-0-1-14"
      >
        Highlight areas of water stress
      </h2>
      <div
        className="square-switch"
        style={
          Object {
            "MozTransition": "opacity 0.25s",
            "MozUserSelect": "none",
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "WebkitTransition": "opacity 0.25s",
            "WebkitUserSelect": "none",
            "borderRadius": 14,
            "direction": "ltr",
            "display": "inline-block",
            "msUserSelect": "none",
            "opacity": 1,
            "position": "relative",
            "textAlign": "left",
            "touchAction": "none",
            "transition": "opacity 0.25s",
            "userSelect": "none",
          }
        }
      >
        <div
          className="react-switch-bg"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "MozTransition": "background 0.25s",
              "WebkitTransition": "background 0.25s",
              "background": "#5f9f7c",
              "borderRadius": 14,
              "cursor": "pointer",
              "height": 28,
              "margin": 0,
              "position": "relative",
              "transition": "background 0.25s",
              "width": 56,
            }
          }
        />
        <div
          className="react-switch-handle"
          onClick={[Function]}
          onMouseDown={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
              "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
              "background": "#ffffff",
              "border": 0,
              "borderRadius": "50%",
              "boxShadow": null,
              "cursor": "pointer",
              "display": "inline-block",
              "height": 26,
              "outline": 0,
              "position": "absolute",
              "top": 1,
              "transform": "translateX(1px)",
              "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
              "width": 26,
            }
          }
        />
        <input
          checked={false}
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyUp={[Function]}
          role="switch"
          style={
            Object {
              "border": 0,
              "clip": "rect(0 0 0 0)",
              "height": 1,
              "margin": -1,
              "overflow": "hidden",
              "padding": 0,
              "position": "absolute",
              "width": 1,
            }
          }
          type="checkbox"
        />
      </div>
    </header>
    <div
      className="LayerCard-content-0-1-15"
    >
      <p
        className="LayerCard-contentDescription-0-1-16"
      />
      <p
        className="LayerCard-source-0-1-19"
      >
        <a
          className="LayerCard-sourceLink-0-1-20"
        />
        <span
          className="LayerCard-sourceDescription-0-1-21"
        />
      </p>
      <div
        style={
          Object {
            "marginBottom": "15px",
          }
        }
      />
    </div>
    <footer
      className="LayerCard-footer-0-1-18"
    >
      <a
        className="LayerCard-moreLink-0-1-17"
        href="/map/learn_more/123"
        onClick={[Function]}
      >
        <i
          className="icon__book-reader"
          style={
            Object {
              "marginRight": 5,
            }
          }
        />
        <span>
          Learn more
        </span>
      </a>
    </footer>
  </div>
  <div
    className="MapSideBar-sideBarContent-0-1-9"
    id="sideBarContent"
  >
    <button
      className="MapSideBar-addLayerButton-0-1-7"
      onClick={[Function]}
      style={
        Object {
          "borderBottomLeftRadius": 0,
          "borderTopLeftRadius": 0,
          "marginLeft": -1,
          "marginTop": 20,
          "minWidth": "45%",
          "paddingLeft": 15,
        }
      }
    >
      <i
        className="icon__plus-circle MapSideBar-addLayerButtonIcon-0-1-8"
      />
      Add datasets
    </button>
  </div>
  <a
    className="map-credit"
    href="//resourcewatch.org/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      src="test-file-stub"
    />
  </a>
</div>
`;
